core/injector/application_injector.cpp

```diff
@@ -4,3 +4,1305 @@
+        di::bind<application::AppConfiguration>.to(config),
+
+        // compose peer keypair
+        di::bind<libp2p::crypto::KeyPair>.to([](auto const &inj) {
+          auto &app_config =
+              inj.template create<const application::AppConfiguration &>();
+          auto &crypto_provider =
+              inj.template create<const crypto::Ed25519Provider &>();
+          auto &crypto_store =
+              inj.template create<const crypto::CryptoStore &>();
+          return get_peer_keypair(app_config, crypto_provider, crypto_store);
+        })[boost::di::override],
+
+        // bind io_context: 1 per injector
+        di::bind<::boost::asio::io_context>.in(
+            di::extension::shared)[boost::di::override],
+
+        di::bind<api::ApiServiceImpl::ListenerList>.to([](auto const
+                                                              &injector) {
+          std::vector<std::shared_ptr<api::Listener>> listeners{
+              injector
+                  .template create<std::shared_ptr<api::HttpListenerImpl>>(),
+              injector.template create<std::shared_ptr<api::WsListenerImpl>>(),
+          };
+          return api::ApiServiceImpl::ListenerList{std::move(listeners)};
+        }),
+        di::bind<api::ApiServiceImpl::ProcessorSpan>.to([](auto const
+                                                               &injector) {
+          static std::vector<std::shared_ptr<api::JRpcProcessor>> processors{
+              injector.template create<
+                  std::shared_ptr<api::state::StateJrpcProcessor>>(),
+              injector.template create<
+                  std::shared_ptr<api::author::AuthorJRpcProcessor>>(),
+              injector.template create<
+                  std::shared_ptr<api::chain::ChainJrpcProcessor>>(),
+              injector.template create<
+                  std::shared_ptr<api::system::SystemJrpcProcessor>>(),
+              injector.template create<
+                  std::shared_ptr<api::rpc::RpcJRpcProcessor>>(),
+              injector.template create<
+                  std::shared_ptr<api::payment::PaymentJRpcProcessor>>()};
+          return api::ApiServiceImpl::ProcessorSpan{processors};
+        }),
+        // bind interfaces
+        di::bind<api::HttpListenerImpl>.to([](const auto &injector) {
+          const application::AppConfiguration &config =
+              injector.template create<application::AppConfiguration const &>();
+          auto app_state_manager =
+              injector.template create<sptr<application::AppStateManager>>();
+          auto context = injector.template create<sptr<api::RpcContext>>();
+          auto &&http_session_config =
+              injector.template create<api::HttpSession::Configuration>();
+
+          return get_jrpc_api_http_listener(
+              config, app_state_manager, context, http_session_config);
+        }),
+        di::bind<api::WsListenerImpl>.to([](const auto &injector) {
```

Comment:
Please, choose one name (**inj** or **injector**) for all cases. For code cleanliness.